server/tests/tag.test.js

```diff
@@ -0,0 +1,45 @@
+import mongoose from 'mongoose';
+import request from 'supertest-as-promised';
+import httpStatus from 'http-status';
+import chai, {expect} from 'chai';
+import app from '../../index';
+
+chai.config.includeStack = true;
+
+/**
+ * root level hooks
+ */
+after((done) => {
+  // required because https://github.com/Automattic/mongoose/issues/1251#issuecomment-65793092
+  mongoose.models = {};
+  mongoose.modelSchemas = {};
+  mongoose.connection.close();
+  done();
+});
+
+xdescribe('## Tag APIs', () => {
+
+  describe('# GET /api/users/:userId', () => {
+    it('should get user details', (done) => {
+      request(app)
+        .get('/api/tags/')
+        .expect(httpStatus.OK)
+        .then((res) => {
+          done();
+        })
+        .catch(done);
+    });
+
+    it('should report error with message - Not found, when tag does not exists', (done) => {
```

Comment:
You need to write a task for the tag detail route as well

Reply:
@TheHollidayInn You mean adding a test for `api/tags/` right? Have done that

Reply:
Nah, tag detail is `api/tags/:tagid`. Looks like you added that though.

Reply:
@TheHollidayInn Did that, removed the wait and fixed the tests. All tests run successfully :)